tests/fsharp/PlatformHelpers.fs

```diff
@@ -0,0 +1,204 @@
+    match runAttempt input with
+    | Success s -> successTrack s
+    | Failure f -> failTrack f
+
+[<DebuggerStepThrough>]
+let bind successTrack = either successTrack failed
+
+[<DebuggerStepThrough>] 
+let fail failTrack result = either succeed failTrack result
+
+[<DebuggerStepThrough>] 
+type Attempt =
+    static member Run x = runAttempt x
+
+[<DebuggerStepThrough>] 
+type AttemptBuilder() =
+    member this.Bind(m : Attempt<_, _>, success) = bind success m
+    member this.Bind(m : Result<_, _>, success) = bind success (fun () -> m)
+    member this.Bind(m : Result<_, _> option, success) = 
+        match m with
+        | None -> this.Combine(this.Zero(), success)
+        | Some x -> this.Bind(x, success)
+    member this.Return(x) : Attempt<_, _> = succeed x
+    member this.ReturnFrom(x : Attempt<_, _>) = x
+    member this.Combine(v, f) : Attempt<_, _> = bind f v
+    member this.Yield(x) = Success x
+    member this.YieldFrom(x) = x
+    member this.Delay(f) : Attempt<_, _> = delay f
+    member this.Zero() : Attempt<_, _> = succeed ()
+    member this.While(guard, body: Attempt<_, _>) =
+        if not (guard()) 
+        then this.Zero() 
+        else this.Bind(body, fun () -> 
+            this.While(guard, body))  
+
+    member this.TryWith(body, handler) =
+        try this.ReturnFrom(body())
+        with e -> handler e
+
+    member this.TryFinally(body, compensation) =
+        try this.ReturnFrom(body())
+        finally compensation() 
+
+    member this.Using(disposable:#System.IDisposable, body) =
+        let body' = fun () -> body disposable
+        this.TryFinally(body', fun () -> 
+            match disposable with 
+                | null -> () 
+                | disp -> disp.Dispose())
+
+    member this.For(sequence:seq<'a>, body: 'a -> Attempt<_,_>) =
+        this.Using(sequence.GetEnumerator(),fun enum -> 
+            this.While(enum.MoveNext, 
+                this.Delay(fun () -> body enum.Current)))
+
+let processor = new AttemptBuilder()
+
+
+let log format = Printf.ksprintf (printfn "%s") format
+
+type OutPipe (mailbox: MailboxProcessor<_>) =
+    member x.Post msg = mailbox.Post(msg)
+    interface System.IDisposable with
+        member x.Dispose () = 
+            async {
+                while mailbox.CurrentQueueLength > 0 do
+                    let timeout = System.TimeSpan.FromMilliseconds(50.0)
+                    do! Async.Sleep (timeout.TotalMilliseconds |> int)
+            } |> Async.RunSynchronously
+
+let redirectTo (writer: TextWriter) =
+    let mailbox = MailboxProcessor.Start(fun inbox -> 
+        let rec loop () = async {
+            let! (msg : string) = inbox.Receive ()
+            do! writer.WriteLineAsync(msg) |> (Async.AwaitIAsyncResult >> Async.Ignore)
+            return! loop () }
+        loop ())
+    new OutPipe (mailbox)
+
+let redirectToLog () = redirectTo System.Console.Out
+
+let inline (/) a (b: string) = System.IO.Path.Combine(a,b)
```

Comment:
In FAKE we use </> as alias for PathCombine.
 
@@ is customized version, which does slightly different things